packages/fxa-auth-server/lib/routes/recovery-key.js

```diff
@@ -297,7 +297,7 @@ module.exports = (
         ...RECOVERY_KEY_DOCS.RECOVERYKEY_EXISTS_POST,
         auth: {
           mode: 'optional',
```

Comment:
Since this now returns the recovery key hint which could contain sensitive data, we'd want to make sure this request always uses either a session token or a passwordForgotToken (= not optional)

Reply:
Good catch.  I think I'll keep it optional for backwards compat but only return a hint if the request's authenticated.